modin/core/dataframe/pandas/exchange/dataframe_protocol/from_dataframe.py

```diff
@@ -0,0 +1,463 @@
+from modin.core.dataframe.base.exchange.dataframe_protocol.utils import (
+    DTypeKind,
+    ColumnNullType,
+)
+from modin.core.dataframe.base.exchange.dataframe_protocol.dataframe import (
+    ProtocolDataframe,
+    ProtocolColumn,
+    ProtocolBuffer,
+)
+
+
+def from_dataframe_to_pandas(
+    df: ProtocolDataframe, allow_copy: bool = True, nchunks: Optional[int] = None
+):
+    """
+    Build ``pandas.DataFrame`` from an object supporting DataFrame exchange protocol (__dataframe__).
+
+    Parameters
+    ----------
+    df : ProtocolDataframe
+        Object supporting the exchange protocol (__dataframe__).
+    allow_copy : bool, default: True
+        Whether to allow for `df` providing a copy of underlying data.
+    nchunks : int, optional
+        Number of chunks to split `df`.
+
+    Returns
+    -------
+    pandas.DataFrame
+    """
+    if not hasattr(df, "__dataframe__"):
+        raise ValueError("`df` does not support __dataframe__")
+
+    df = df.__dataframe__()
+    if isinstance(df, dict):
+        df = df["dataframe"]
+
+    def get_pandas_df(df):
+        # We need a dict of columns here, with each column being a numpy array (at
+        # least for now, deal with non-numpy dtypes later).
+        columns = dict()
+        buffers = []  # hold on to buffers, keeps memory alive
+        for name in df.column_names():
+            if not isinstance(name, str):
+                raise ValueError(f"Column {name} is not a string")
+            if name in columns:
+                raise ValueError(f"Column {name} is not unique")
+            col = df.get_column_by_name(name)
+            dtype = col.dtype[0]
+            if dtype in (
+                DTypeKind.INT,
+                DTypeKind.UINT,
+                DTypeKind.FLOAT,
+                DTypeKind.BOOL,
+            ):
+                columns[name], buf = convert_primitive_column_to_ndarray(col)
+            elif dtype == DTypeKind.CATEGORICAL:
+                columns[name], buf = convert_categorical_column(col)
+            elif dtype == DTypeKind.STRING:
+                columns[name], buf = convert_string_column(col)
+            elif dtype == DTypeKind.DATETIME:
+                columns[name], buf = convert_datetime_column(col)
+            else:
+                raise NotImplementedError(f"Data type {dtype} not handled yet")
+
+            buffers.append(buf)
+
+        pandas_df = pandas.DataFrame(columns)
+        pandas_df._buffers = buffers
+        return pandas_df
+
+    pandas_dfs = []
+    for chunk in df.get_chunks(nchunks):
+        pandas_df = get_pandas_df(chunk)
+        pandas_dfs.append(pandas_df)
+
+    pandas_df = pandas.concat(pandas_dfs, axis=0, ignore_index=True)
+
+    index_obj = df.metadata.get("modin.index", df.metadata.get("pandas.index", None))
+    if index_obj is not None:
+        pandas_df.index = index_obj
+
+    return pandas_df
+
+
+def convert_primitive_column_to_ndarray(col: ProtocolColumn) -> Tuple[np.ndarray, Any]:
```

Comment:
Let's align names of the functions, which convert smth to smth.
1. convert_smth_to_smth(...)
2. smth_to_smth(...)
3. other thoughts?

Reply:
+ for 2

Reply:
applied second suggestion